interfaces/builtin/utils.go

```diff
@@ -110,7 +111,20 @@ func aareExclusivePatterns(orig string) []string {
 // dirs.SnapDesktopFilesDir, but explicitly denies access to all other snaps'
 // desktop files since xdg libraries may try to read all the desktop files
 // in the dir, causing excessive noise. (LP: #1868051)
-func getDesktopFileRules(snapInstanceName string) []string {
+func getDesktopFileRules(snapInstanceName string, spec *apparmor.Specification) []string {
+	// desktop-launch allows to read all .desktop files; but "deny" rules overrule any "allow"
+	// rule, so we must not add these rules if this snap uses the desktop-launch interface.
+	// Also, for security reasons, all these rules are removed if the desktop-launch interface
+	// is listed, thus only if it is really connected will the snap have any kind of access to
+	// these folders/files.
+	if spec != nil {
+		for _, plug := range spec.SnapAppSet().Info().Plugs {
+			if plug.Interface == "desktop-launch" {
```

Comment:
So I've been looking at this and I feel a bit stupid as my memory of this area is somewhat rusty but let me spell out my concern.

Is there anything changed, when `desktop-lauch` is disconnected, and we call `getDesktopFileRules` from `AppArmorConnectedPlug` in unity7.go?

We are looking at `*snap.Info`'s `Plugs` attribute which knows nothing about connections.

If we are happy with this specific semantics then can we please add a comment with a big FIXME to make clear this is going on before merging this?

CC @pedronis

Reply:
No, nothing changes. It only checks if the snap also requests the plug, but not if it's already connected or not. It's not a problem because without the plug connected, the net result in this specific case is less privileges than with the plug connected, AND less privileges than without the plug declared, thus any snap that requests `desktop-launch` interface will have less privileges than the same snap without that requests.

Of course, this is an ugly hack; that's why I'm already working on a more generic mechanism to replace this code.

I'll add the FIXME comment.

Reply:
Added.